domain/user/service/service.go

```diff
@@ -19,46 +20,46 @@ import (
 type State interface {
 	// AddUser will add a new user to the database. If the user already exists
 	// an error that satisfies usererrors.AlreadyExists will be returned. If the
-	// users creator is set and does not exist then a error that satisfies
+	// users creator is set and does not exist then an error that satisfies
 	// usererrors.UserCreatorNotFound will be returned.
-	AddUser(context.Context, user.User) error
+	AddUser(context.Context, string, user.User, string) error
```

Comment:
These `string` types for the uuid should all be users.UUID now ?

Reply:
Switch where it was reasonable.